metadata/snapshot.go

```diff
@@ -326,6 +326,10 @@ func (s *snapshotter) createSnapshot(ctx context.Context, key, parent string, re
 			return err
 		}
 
+		if err := addSnapshotLease(ctx, tx, s.name, key); err != nil {
```

Comment:
I am studying the lease. I have a question.
When we call sn.Prepare,we add the key to lease bucket. 
On the time, the key in snapshot bucket agree with it in lease bucket.
But the key  is not chainID.  E.g.  It is not chainID but "extract-%s %s" in ApplyLayer function.

When call sn.Commit, the key in the snapshot bucket be changed to chainID, but we don't change the key which is in lease bucket.
Is there a window where the gc will collect the snapshot.
@dmcgowan  PTAL
```
func ApplyLayer(ctx context.Context, layer Layer, chain []digest.Digest, sn snapshot.Snapshotter, a diff.Differ
         key := fmt.Sprintf("extract-%s %s", uniquePart(), chainID)

	// Prepare snapshot with from parent, label as root
	mounts, err := sn.Prepare(ctx, key, parent.String(), opts...)
  
        if err = sn.Commit(ctx, chainID.String(), key, opts...); err != nil {
```

Reply:
@yanxuean you are correct, we should be adding the committed objects to the lease as well. Committed snapshots are usually protected by being referenced by images, containers, or other snapshots, however you are right to assume that there is a window before the references are created in which these could be GC'ed if they are not added to the lease. I will update this as part of another lease PR I am working on. Thanks for reviewing the code!